### PR TITLE
Chaos tests for JS cluster

### PR DESCRIPTION
**Pull request by mprimi**
- [ ] Link to issue, e.g. `Resolves #NNN`
 - [x] Documentation added (if applicable)
 - [x] Tests added
 - [x] Branch rebased on top of current main (`git pull --rebase origin main`)
 - [x] Changes squashed to a single commit ** => intentionally split in 2 commits for easier review**
 - [x] Build is green in Travis CI
 - [x] You have certified that the contribution is your original work and that you license the work to the project under the [Apache 2 license](https://github.com/nats-io/nats-server/blob/main/LICENSE)

### Changes proposed in this pull request:

- Add chaos controller
- Add "cluster bouncer" chaos monkey
- Add 2 tests that use it
- Add some utility methods

/cc @nats-io/core

### CODE REVIEW

File: scripts/runTestsOnTravis.sh [hunk 1]
```diff
@@ -51,6 +51,13 @@ elif [ "$1" = "js_super_cluster_tests" ]; then
 
     go test -race -v -run=TestJetStreamSuperCluster ./server -count=1 -vet=off -timeout=30m -failfast
 
+elif [ "$1" = "js_cluster_chaos_tests" ]; then
+
+    # Run JetStream clustered chaos tests. By convention, all JS cluster chaos
+    # tests with `TestJetStreamClusterChaos`.
+
+    go test -race -v -run=TestJetStreamClusterChaos ./server -count=1 -vet=off -timeout=30m -failfast
```
Comment (kozlovic) [inline scripts/runTestsOnTravis.sh:59]: Either name this `TestJetStreamChaosCluster` or then you need to add `skip_js_cluster_chaos_tests` to the "js_cluster_tests" section above...

File: server/jetstream_chaos_helpers_test.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
+// Copyright 2020-2022 The NATS Authors
```
Comment (kozlovic) [inline server/jetstream_chaos_helpers_test.go:1]: Since this is a new file, `2022` is enough

File: server/jetstream_chaos_helpers_test.go [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+import (
+	"math/rand"
+	"sync"
+	"testing"
+	"time"
+)
+
+// Support functions
+
+func init() {
```
Comment (kozlovic) [inline server/jetstream_chaos_helpers_test.go:25]: Would remove that since it is empty?

File: server/jetstream_cluster_chaos_test.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
+// Copyright 2020-2022 The NATS Authors
```
Comment (kozlovic) [inline server/jetstream_cluster_chaos_test.go:1]: Same

**Review by kozlovic — CHANGES_REQUESTED** [review]
Check for my comments and notice some issues on Travis regarding fmt or unused imports. Also, may be worth (at least to experiment), add an entry in ".travis.yml" to actually trigger the tests, just to make sure that all is setup properly for that.

**Review by mprimi — COMMENTED** [review]
Thank you for the review @kozlovic 

All comments addressed.

**Review by kozlovic — CHANGES_REQUESTED** [review]
Still have some issues, such as:
```
# github.com/nats-io/nats-server/v2/server [github.com/nats-io/nats-server/v2/server.test]
332server/jetstream_chaos_helpers_test.go:17:2: imported and not used: "fmt"
333server/jetstream_chaos_helpers_test.go:18:2: imported and not used: "log"
334server/jetstream_chaos_helpers_test.go:20:2: imported and not used: "strings"
```

**Review by kozlovic — CHANGES_REQUESTED** [review]
What about adding this to .travis.yml, at least to make sure that they run correctly (even if we were to remove/comment out once it is time to merge if we want to not run them all the time):
```
    - name: "Run JetStream chaos tests"
      env: TEST_SUITE=js_chaos_tests
```
You could put that say below the line of the cluster tests `env: TEST_SUITE=js_cluster_tests`

**Comment by mprimi** [comment]
@kozlovic 
> What about adding this to .travis.yml, 

I wasn't sure about adding this to the PR build critical path. Some of the tests are failing, and I need to set a large timeout once I add them.

Anyway, added now, we can always remove it later.

**Comment by kozlovic** [comment]
@mprimi I just wanted your to add a new commit with that to make sure that it would trigger and run at least to make sure everything was ok. Yeah, seeing that it is 14 minutes long (just for 2 tests), maybe we don't include them by default in Travis, but devs will have options to run them locally (with naming convention). So please remove from travis.yml now that we know everything is setup correctly.

**Comment by mprimi** [comment]
@kozlovic, gotch. Removed from travis

**Review by kozlovic — APPROVED** [review]
LGTM

**Pull request merged**